clang/test/SemaCXX/gh102293.cpp

```diff
@@ -20,3 +19,15 @@ struct S : HasVT {
   HasD<> v;
 };
 
+// Ensure we don't get infinite recursion from the check, however. See GH104802
+namespace GH104802 {
+class foo {       // expected-note {{definition of 'GH104802::foo' is not complete until the closing '}'}}
+  foo a;          // expected-error {{field has incomplete type 'foo'}}
```

Comment:
Should we also text cv qualified version of the type as well?

Reply:
Yes, absolutely! I'll add coverage (and probably fix a bug).